spark/src/main/java/org/opensearch/sql/spark/flint/operation/FlintIndexOpAlter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.flint.operation;
+
+import lombok.SneakyThrows;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.spark.client.EMRServerlessClient;
+import org.opensearch.sql.spark.dispatcher.model.FlintIndexOptions;
+import org.opensearch.sql.spark.execution.statestore.StateStore;
+import org.opensearch.sql.spark.flint.FlintIndexMetadata;
+import org.opensearch.sql.spark.flint.FlintIndexMetadataService;
+import org.opensearch.sql.spark.flint.FlintIndexState;
+import org.opensearch.sql.spark.flint.FlintIndexStateModel;
+
+/**
+ * Index Operation for Altering the flint index. Only handles alter operation when
+ * auto_refresh=false.
+ */
+public class FlintIndexOpAlter extends FlintIndexOp {
+  private static final Logger LOG = LogManager.getLogger(FlintIndexOpAlter.class);
+  private final EMRServerlessClient emrServerlessClient;
+  private final FlintIndexMetadataService flintIndexMetadataService;
+  private final FlintIndexOptions flintIndexOptions;
+
+  public FlintIndexOpAlter(
+      FlintIndexOptions flintIndexOptions,
+      StateStore stateStore,
+      String datasourceName,
+      EMRServerlessClient emrServerlessClient,
+      FlintIndexMetadataService flintIndexMetadataService) {
+    super(stateStore, datasourceName);
+    this.emrServerlessClient = emrServerlessClient;
+    this.flintIndexMetadataService = flintIndexMetadataService;
+    this.flintIndexOptions = flintIndexOptions;
+  }
+
+  @Override
+  protected boolean validate(FlintIndexState state) {
+    return state == FlintIndexState.ACTIVE || state == FlintIndexState.REFRESHING;
```

Comment:
FlintIndexState.ACTIVE is validate state?

Reply:
Before altering the index should be either active or refreshing? Do you see any issue with this?

Reply:
Any case index is active, but auto_refresh=false, only for retry case?